projector-plugin/src/main/kotlin/org/jetbrains/projector/plugin/ui/ProjectorStatusWidget.kt

```diff
@@ -96,23 +91,45 @@ class ProjectorStatusWidget(project: Project)
     }
   }
 
-  private fun updateText() = "Projector"
+  private fun updateText() = "Projector" //  (${clientsCount()})"
 
   private fun updateTooltip(): String {
     return when {
       isActivationNeeded() -> "Activation is needed"
-      isProjectorRunning() -> "Projector is running"
+      isProjectorRunning() -> {
+        "Projector is running" //. Connected client(s): ${clientsCount()}"
+      }
       isProjectorAutoStarting() -> "Projector is starting"
-      isProjectorDisabled() -> "Projector is disabled"
+      isProjectorDisabled() -> "Projector is down"
+      isHeadlessProjectorDetected() -> "Headless projector detected, plugin is disabled"
       else -> "Impossible state"
     }
   }
 
-  private companion object {
-    private const val PROJECTOR_TOOLBAR_ACTION_GROUP = "projector.toolbar"
+  private fun onClick() {
+    when {
+      isActivationNeeded() -> fireAction(ActivateAction.ID)
+      isProjectorRunning() -> fireAction(SessionAction.ID)
+      isProjectorAutoStarting() -> fireAction(WaitForStartAction.ID)
+      isProjectorDisabled() -> fireAction(EnableAction.ID)
+      isHeadlessProjectorDetected() -> fireAction(HeadlessProjectorAction.ID)
+    }
+  }
+
+  private fun fireAction(actionId: String) {
+    val am = ActionManager.getInstance()
+    val action = am.getAction(actionId)
+    val ctx = DataManager.getInstance().getDataContext(myStatusBar as Component)
+    val event = AnActionEvent.createFromAnAction(action, null, "", ctx)
+    action.actionPerformed(event)
+  }
+
+  companion object {
+    val ID: String by lazy { ProjectorStatusWidget::class.java.name }
     private val RED_DOT: Icon by lazy { IconLoader.getIcon("/META-INF/redSign.svg", ProjectorStatusWidget::class.java) }
     private val GREEN_DOT: Icon by lazy { IconLoader.getIcon("/META-INF/greenSign.svg", ProjectorStatusWidget::class.java) }
     private val YELLOW_DOT: Icon by lazy { IconLoader.getIcon("/META-INF/yellowSign.svg", ProjectorStatusWidget::class.java) }
     private val BLACK_DOT: Icon by lazy { IconLoader.getIcon("/META-INF/blackSign.svg", ProjectorStatusWidget::class.java) }
+    private fun clientsCount() = if (isProjectorRunning()) ProjectorService.getClientList().size else 0
```

Comment:
1. Unused.
2. Does it make sense to return 0 if the server isn't running? It depends on how we plan to use it, maybe `null` is more appropriate.

Reply:
Fixed

Reply:
Fixed problem with widget in dumb mode.